index.js

```diff
@@ -12,16 +16,24 @@ module.exports = (stream, goodbye) => {
     sink: source => stream.sink((async function * () {
       // when the goodbye is received, allow the source to end.
       for await (const chunk of source) {
-        const buff = Buffer.from(chunk)
-        const done = buff.slice(-goodbye.length).equals(goodbye)
-        if (done) {
-          const remaining = buff.length - goodbye.length
-          if (remaining > 0) {
-            yield buff.slice(0, remaining)
+        if (isBufferCompatible(chunk)) {
```

Comment:
Is it reasonable to assume that all chunks in the source are buffer compatible if `goodbye` is buffer compatible?

It would be nice to not have to do the check every iteration - I think a stream of buffers will never switch to objects mid way and vice versa.